### ISSUE TITLE
[4.0] Unify javascript

### ISSUE THREAD

**Issue by dgrammatiko** [joomla/joomla-cms#20397]
### Web components and Custom elements are also ES6

Since we already have a PR for merging the ongoing work from the ES6-javascript group to the main repo I would like to propose one more change for the shake of consistency but also (IMHO) simplifying  the contributors workflow:

Unify the source code of the ES6 (plain) files with those of the web components/custom elements. Right now web components source is living in the `/build/webcomponents` folder. So my proposal will be to expose another root directory, eg `media_src` with all the source files (js and scss but also images and have one more method to smash them down, already plenty of such scripts in the npmjs). This in combination to our ultimate goal having **only** source in the main repo will unify things (in plain english I'm proposing to build the whole media folder through our existing/improved tools, already just one simple line on the terminal: `npm install`).

**Comment by laoneo** [comment]
Sounds like a good plan. But then I would name the the source folder assets_src to reflect better what it actually is.

**Comment by mbabker** [comment]
Put it in the build folder.

We don't need another top level directory in the repository.
